gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -103,21 +112,39 @@ private Set<GroupPartOrPackageName> fetchFoldersFromContent(String contents) {
     }
 
     private Set<DependencyVersion> parseVersionsFromContent(String content) {
-        Set<DependencyVersion> versions = new HashSet<>();
         try {
             XmlMapper xmlMapper = new XmlMapper();
+            xmlMapper.registerModule(new GuavaModule());
 
             Metadata metadata = xmlMapper.readValue(content, Metadata.class);
-            if (metadata.versioning() != null && metadata.versioning().versions() != null) {
-                String latest = metadata.versioning().latest();
-                for (String version : metadata.versioning().versions()) {
-                    versions.add(DependencyVersion.of(version, latest.equals(version)));
-                }
+            List<String> allVersions = new ArrayList<>(metadata.versioning().versions());
+            Collections.reverse(allVersions);
+
+            if (allVersions.isEmpty()) {
+                return Collections.emptySet();
             }
+
+            String latest = Optional.ofNullable(metadata.versioning().release())
+                    .filter(l -> !l.isEmpty())
+                    .orElseGet(() -> metadata.versioning().latest());
+
+            if (VERSION_PATTERN.matcher(latest.toLowerCase(Locale.ROOT)).matches()) {
```

Comment:
let's extract `VERSION_PATTERN.matcher(...).matches()` to a method:

```java
boolean isVersion(String maybeVersion) {
  return VERSION_PATTERN.matcher(latest.toLowerCase(Locale.ROOT)).matches();
}
```

then replace both usages of it